.github/workflows/haskell.yml

```diff
@@ -11,63 +11,48 @@ permissions:
 jobs:
   build:
 
-    runs-on: ${{ matrix.os }}
+    runs-on: ${{ matrix.sys.os }}
 
     strategy:
       fail-fast: false
       matrix:
-        ghc: ["8.10.7", "9.2.7"]
-        cabal: ["3.8.1.0"]
-        os: [ubuntu-latest, windows-latest, macOS-latest]
+        ghc: ["8.10.7", "9.2.8", "9.4.5", "9.6.2"]
+        cabal: ["3.10.1.0"]
+        sys:
+          - { os: windows-latest, shell: 'C:/msys64/usr/bin/bash.exe -e {0}' }
+          - { os: ubuntu-latest, shell: bash }
+          - { os: macos-latest, shell: bash}
 
-    steps:
-    - name: Checkout repository
-      uses: actions/checkout@v3
-
-    - name: "Linux: Install system dependencies (apt-get)"
-      if: runner.os == 'Linux'
-      run: |
-        sudo apt-get update
-        sudo apt-get -y install liblmdb-dev
-        sudo apt-get -y autoremove
-
-    - name: "MacOS: Install system dependencies (brew)"
-      if: runner.os == 'macOS'
-      run: |
-        brew install pkg-config lmdb
-        cp ./.github/workflows/lmdb.pc /usr/local/lib/pkgconfig
-
-    - name: "Windows: Install system dependencies via pacman (msys2)"
-      if: runner.os == 'Windows'
-      shell: pwsh
-      run: |
-        ghcup run -- pacman --noconfirm -S `
-          mingw-w64-x86_64-pkg-config `
-          mingw-w64-x86_64-lmdb
-
-    - name: "Windows: Update PATH"
-      if: runner.os == 'Windows'
-      run: |
-        $env:PATH=("C:\msys64\mingw64\bin;{0}" -f $env:PATH)
-        echo "PATH=$env:PATH" >> $env:GITHUB_ENV
+    defaults:
+        run:
+          shell: ${{ matrix.sys.shell }}
 
+    steps:
     - name: Setup Haskell
       id: setup-haskell
-      uses: haskell/actions/setup@v2
+      uses: input-output-hk/actions/haskell@latest
       with:
         ghc-version: ${{ matrix.ghc }}
         cabal-version: ${{ matrix.cabal }}
-        cabal-update: false
 
-    - name: "Linux/MacOS: Configure cabal.project.local"
-      if: runner.os != 'Windows'
-      run: |
-        cp .github/workflows/cabal.project.local.Linux cabal.project.local
+    - name: Checkout repository
+      uses: actions/checkout@v3
 
-    - name: "Windows: configure cabal.project.local"
-      if: runner.os == 'Windows'
+    - name: "Install system dependencies"
+      if: runner.os == 'Linux'
+      run: |
+        if [[ ${{ runner.os }} == 'Linux' ]]; then
+          sudo apt-get -y install liblmdb-dev
+        elif [[ ${{ runner.os }} == 'macOS' ]]; then
+          brew install lmdb
+          cp ./.github/workflows/lmdb.pc /usr/local/lib/pkgconfig
+        else
+          /usr/bin/pacman --noconfirm -S mingw-w64-x86_64-lmdb
+        fi
```

Comment:
This will only run on `Linux`, and I also think it's slightly clearer if we have a separate step for each OS